proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -239,6 +240,13 @@ public void fireAndForget() {
             connect()
                     .whenComplete((status, throwable) -> {
                         if (throwable != null) {
+                            if (throwable instanceof ConnectException) {
+                                Optional<ServerInfo> nextServer = getNextServerToTry();
+                                if (nextServer.isPresent()) {
+                                    createConnectionRequest(nextServer.get()).fireAndForget();
```

Comment:
This is problematic: This would trigger even when the player was trying to connect to another server. Can you add in a check to ensure that this could only be triggered when the player is not yet connected to another server?

Additionally, I have concerns about this logic being in `fireAndForget()` - this is going to cover only connection exceptions, not kicks from servers.

Reply:
I have pushed a new commit which would check that the player is not currently connected to a server. However, as for relocating it, I'm unsure as to a better place to put it as the exception is handled in `fireAndForget()`. If by you mean "not kicks from servers" you mean to handle if a player is kicked, move then back to a default server, I could do that.

Reply:
it's more during connection, the two major reasons why you'd wanna try the next server is because a connection error occured, or because the server kicked during login for some reason, e.g. "wrong client version"

Reply:
I would move this logic to #handleConnectionException(ServerInfo info, Component disconnectReason)

Reply:
@hugmanrique is right here. You should do this in the `handleConnectionException` method. It's called any time there's a disconnect from the server and it's the best place to put this code.